arch/x86/core/cet.c

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2025 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/**
+ * @file
+ * @brief Indirect Branch Tracking
+ *
+ * Indirect Branch Tracking (IBT) setup routines.
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/arch/x86/msr.h>
+#include <zephyr/arch/x86/cet.h>
+#include <zephyr/logging/log_ctrl.h>
+#include <zephyr/logging/log.h>
+
+#include <kernel_arch_data.h>
+LOG_MODULE_DECLARE(os, CONFIG_KERNEL_LOG_LEVEL);
+
+#ifdef CONFIG_X86_64
+#define TOKEN_OFFSET 5
+#else
+#define TOKEN_OFFSET 4
+#endif
+
+#ifdef CONFIG_HW_SHADOW_STACK
+#ifdef CONFIG_HW_SHADOW_STACK_ALLOW_REUSE
+extern void arch_shadow_stack_reset(k_tid_t thread);
+#endif /* CONFIG_HW_SHADOW_STACK_ALLOW_REUSE */
+
+int arch_thread_hw_shadow_stack_attach(k_tid_t thread,
+				       arch_thread_hw_shadow_stack_t *stack,
+				       size_t stack_size)
+{
+	/* Can't attach to NULL */
+	if (stack == NULL) {
+		LOG_ERR("Can't set NULL shadow stack for thread %p\n", thread);
+		return -EINVAL;
+	}
+
+	/* Or if the thread already has a shadow stack. */
+	if (thread->arch.shstk_addr != NULL) {
+#ifdef CONFIG_HW_SHADOW_STACK_ALLOW_REUSE
+		/* Allow reuse of the shadow stack if the base and size are the same */
+		if (thread->arch.shstk_base == stack ||
```

Comment:
```suggestion
		if (thread->arch.shstk_base && stack ||
```

Reply:
I'm not sure I've got the suggestion here... I'm trying to prevent someone from changing the shadow stack pointer of a thread here, so I compare the new argument with current value (setting to the same value is a normal thread re-creation operation). What am I missing?

Reply:
Suggestion is to align with the comment on the line above (or fix the comment I guess ;))

Reply:
Also. suggestion has just been edited since I just realized it had a typo which was indeed pretty confusing... 😵 Sorry!

Reply:
Ah, got it. Fixed. Thanks!